code/scripting/api/objs/ship.cpp

```diff
@@ -1649,6 +1649,34 @@ ADE_FUNC(triggerSubmodelAnimation, l_Ship, "string type, string triggeredBy, [bo
 	return Ship_info[shipp->ship_info_index].animations.parseScripted(model_get_instance(shipp->model_instance_num), animtype, trigger).start(forwards ? animation::ModelAnimationDirection::FWD : animation::ModelAnimationDirection::RWD, forced || instant, instant, pause) ? ADE_RETURN_TRUE : ADE_RETURN_FALSE;
 }
 
+ADE_FUNC(setAnimationSpeed, l_Ship, "string type, string triggeredBy, [number speedMultiplier = 1.0]",
+	"Sets the speed multiplier at which an animation runs. Anything other than 1 will not work in multiplayer. Type is the string name of the animation type, "
```

Comment:
Because mission time is synced now, there is probably a way to transmit an update to the speed and have the client sync the rotation, as long as mission time from `Multi_timing_info._current_time` in `multi_time_manager.cpp` is sent with it.  Any other change to animations can probably be handled the same way.

That's for the future, of course, assuming anyone asks for it.

Reply:
Yes. Eventually, I want this and the new instance animation flags to sync.
Trouble with syncinc speed is the use case. I expect this value to be updated every frame in some uses to make the animation speed (e.g. of walker legs) match the ship speed. Hence sending things each frame, especially in larger values, could lead to serious desync. So even if we sync this speed value, we'll also need some mechanism that occasionally helps to resync animations over time. But yeah, that is definitely for the future and will need careful consideration

Reply:
Dropped packets are the enemy here.... Rare as they are.

Reply:
Absolutely